tensorflow/compiler/mlir/lite/transforms/optimize_patterns.td

```diff
@@ -760,17 +760,15 @@ foreach ArgMinMaxOp = [TFL_ArgMinOp, TFL_ArgMaxOp] in {
      (AxesIsLastDimension $axes, $logits)]>;
 }
 
-class AllElementsAreI64<string val> : Constraint<CPred<
-  "($0.isa<DenseElementsAttr>() && "
-   "$0.cast<DenseElementsAttr>().getType().cast<ShapedType>().getElementType().isInteger(64) && "
-   "std::all_of($0.cast<DenseElementsAttr>().getValues<int64_t>().begin(), "
-               "$0.cast<DenseElementsAttr>().getValues<int64_t>().end(), "
-               "[](int64_t v){ return v == " #val# ";}))">>;
+class AllElementsAreInt<string val> : Constraint<CPred<
+  "($0.isa<DenseIntElementsAttr>() && "
+  "llvm::all_of($0.cast<DenseIntElementsAttr>().getIntValues(), "
+               "[](APInt v){ return v.getSExtValue() == static_cast<int64_t>(" #val# ");}))">>;
 
 // Remove Slice ops slicing the whole input tensor, effectively no-op
 def OptimizeSliceOp : Pat<
   (TFL_SliceOp:$output $input, (ConstantOp $begin), $shape),
   (replaceWithValue $input),
-  [(AllElementsAreI64<"0"> $begin),
+  [(AllElementsAreInt<"0"> $begin),
```

Comment:
can we limit only int32 and int64 inputs?

Reply:
Of course we can, but we have already had type constraint in ODS so begin will be either i32 or i64, not sure if it will be a redundant check?

Reply:
This optimization pattern will be applied after TF -> TFL legalization. It is better to check the types in here.

Reply:
Got it! Updated 😃